src/Sfa.Tl.ResultsAndCertification.Web/Controllers/TrainingProviderController.cs

```diff
@@ -471,12 +472,17 @@ public async Task<IActionResult> SearchLearnerRecordNotFoundAsync()
         public async Task<IActionResult> LearnerRecordDetailsAsync(int profileId)
         {
             var viewModel = await _trainingProviderLoader.GetLearnerRecordDetailsAsync<LearnerRecordDetailsViewModel>(User.GetUkPrn(), profileId);
+
             if (viewModel == null || !viewModel.IsLearnerRegistered)
             {
                 _logger.LogWarning(LogEvent.NoDataFound, $"No learner record details found or learner is not registerd or learner record not added. Method: LearnerRecordDetailsAsync({User.GetUkPrn()}, {profileId}), User: {User.GetUserEmail()}");
                 return RedirectToRoute(RouteConstants.PageNotFound);
             }
 
+            var assessment = await _assessmentSeriesLoader.GetResultCalculationAssessmentAsync();
```

Comment:
Why don't you just call the API from the TrainingProviderLoader to get the assesment and load the viewModel there?